README.md

```diff
@@ -109,24 +109,28 @@ A profile can be identified by any combination of the following:
 
 Identifiers are persisted to local storage so that the SDK can keep track of the current profile.
 
-Profile identifiers and other attributes can be set all at once as a `map`:
+Profile identifiers and other attributes can be set all at once using the `Profile` data class:
 
 ```kotlin
 val profile = Profile(
-    mapOf(
-        ProfileKey.EMAIL to "kermit@example.com",
-        ProfileKey.FIRST_NAME to "Kermit"
+    externalId = "USER_IDENTIFIER",
+    email = "kermit@example.com",
+    phoneNumber = "+12223334444",
+    properties = mapOf(
+        ProfileKey.FIRST_NAME to "Kermit",
+        ProfileKey.CUSTOM("instrument") to "banjo"
```

Comment:
should so also update the `setProperty` setter to look for Email, Phone, ExternalID keys and route them to the dedicated setters?

Reply:
Doesn't make much difference, dedicated setter still just does `this[ProfileKey.PHONE_NUMBER] = value`. I guess other than that you could set them to any `serializable`, not just `String`